README.md

```diff
@@ -1,2 +1,161 @@
+Each message will be displayed separately and blocks side-panel until player clicks  
+Can contains also interpolated variables - `[:var_name:]`
+
+Menus:
+
+```
++ First choice:
+	What to do if player click at First choice
++ Second choice:
+	What to do if player click at First choice
+```
+
+Menu blocks all space while on screen
+
+Actions:
+
+```
+> action name:
+	Some block of text or commands
+	.
+```
+
+In any block you can get current active action from variable `action`:  
+`0` - look  
+`1` - interact  
+`2` - apply something from the inventory  
+Also variables `LOOK`, `INTERACT` and `USE` set to `0` or `1` accordingly to variable `action`  
+Any action must ends up with `.` - analog of `return` in programming languages, because action called by `call`, not `jump` (detailed information below)
+
+Locations:
+
+`[location name]` or `[location name (display name)]`  
+It’s like any actions, but also clear screen and set title in GUI  
+In sum, actions and locations are just labes in code.
+
+Items:
+
+You can place some item in the inventory by `+{name}` or `+{name (display name)}`  
+Or you can remove it from inventory by `-{name}`  
+You can also clear the entire inventory by `-{}`  
+All inventory items are displayed in side panel and click on the item is equal click on action with the same name
+
+Variables:
+
+`var_name = expression` -  set the variable value by some expression  
+Values can be only integers.  
+Value of uninitialized variable is `0`  
+Instead of variables in expressions you can use `{item}` - it’ll return `1` if this item stored in your inventory, or `2` if this item is also selected for interaction
+
+Operators:
+
+Unary
+
+| - |  |
+| --- | --- |
+| : | abs |
+| ~ | comp |
+| ! | not |
+| @ | rand (@3 can return 0, 1 or 2) |
+
+Binary
+
+| + | add |
+| --- | --- |
+| - | sub |
+| * | mul |
+| / | div |
+| % | mod |
+| ** | pow |
+
+| && | and |
+| --- | --- |
+| || | or |
+
+| & | band |
+| --- | --- |
+| | | bor |
+| ^ | xor |
+| << | lsh |
+| >> | rsh |
+
+| > | gt |
+| --- | --- |
+| >= | ge |
+| < | lt |
+| <= | le |
+| == | eq |
+| != | ne |
+
+| <<= | min |
+| --- | --- |
+| >>= | max |
+
+Control flow:
+
+`jump location or action name` - jump to location or action  
+`call location or action name` - call location or action as function  
+Jump operator will just change current code position, call operator also add current address to call-stack.  
+Game starts from label `start`
+
+Conditions:
+
+```
+if some_statement:
+	what to do if statement is true
+else:
+	what to do if statement is false
+
+if some_statement:
+	conditin without else
+```
+
+Custom commands:
+
+`// command` - any string after `//` will be passed to `on_command` callback
+
+Other commands:
+
+`…` - game will be wait for players click  
+`.` - stop executing script  
+If call stack is empty, script will just stop executing, otherwise pop address from stack and go to this address.  
+`[]` - clear screen
```

Comment:
```suggestion
`…` - game will wait for player's click  
`.` - stop script's execution
If call stack is empty, script just stops execution, otherwise pops an address from stack and goes to this address.  
`[]` - clear screen
```